app/waf/src/controllers/declaration.controller.ts

```diff
@@ -408,11 +415,13 @@ export class DeclarationController extends BaseController {
 
         try {
           let as3Request = new AS3DeployRequest(adc, application, declaration);
-          await proxyMgr.deploy(
-            mgmt.ipAddress,
-            mgmt.tcpPort,
-            as3Request.declaration,
+          let declBody = as3Request.declaration;
+
+          await this.assertVirtualAddressAvailability(adc, declBody).then(() =>
```

Comment:
one possible proposal: how about bindVirtualAddressToExt() directly, and catch the possible exception from neutron, if the addr is indeed occupied already. (instead of using the heavy assertVirtualAddressAvailability).

Reply:
yeah, binding directly is better.